client/packages/system/src/Immunisation/ListView/ListView.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, useState } from 'react';
+import {
+  TableProvider,
+  DataTable,
+  useColumns,
+  useUrlQueryParams,
+  useNavigate,
+  NothingHere,
+  useTranslation,
+  createTableStore,
+  createQueryParamsStore,
+  useEditModal,
+} from '@openmsupply-client/common';
+import { Toolbar } from './Toolbar';
+import { AppBarButtons } from './AppBarButtons';
+import { ImmunisationProgramCreateModal } from './ImmunisationProgramCreateModal';
+
+export interface Program {
+  id: string;
+  name: string;
+  immunisations: string[];
+  isNew: boolean;
+}
+
+const ProgramListComponent: FC = () => {
+  const {
+    updateSortQuery,
+    updatePaginationQuery,
+    queryParams: { sortBy, page, first, offset },
+  } = useUrlQueryParams({ filters: [{ key: 'name' }] });
+  const pagination = { page, first, offset };
+  const navigate = useNavigate();
+  const t = useTranslation('catalogue');
+
+  const draftPrograms: Record<string, Program> = {};
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [draft] = useState<Record<string, Program>>(draftPrograms);
```

Comment:
I don't think we need this approach of creating multiple drafts. We can just create a single one, then refresh the list to display them all. Don't need this approach (probably was needed for the Demographics but not here.

Something more like how we do assets would be more consistent here I think.

**List view**
https://github.com/msupply-foundation/open-msupply/blob/eee19c957faef357f105aeec1ae04afbfa414300/client/packages/coldchain/src/Equipment/ListView/ListView.tsx#L41
https://github.com/msupply-foundation/open-msupply/blob/eee19c957faef357f105aeec1ae04afbfa414300/client/packages/coldchain/src/Equipment/ListView/ListView.tsx#L147

**Create Modal**
https://github.com/msupply-foundation/open-msupply/blob/develop/client/packages/coldchain/src/Equipment/ListView/CreateAssetModal.tsx